modelql-core/src/commonMain/kotlin/org/modelix/modelql/core/IStep.kt

```diff
@@ -200,6 +200,12 @@ abstract class MonoTransformingStep<In, Out> : TransformingStep<In, Out>(), IMon
     override fun evaluate(evaluationContext: QueryEvaluationContext, queryInput: Any?): Optional<Out> {
         return getProducer().evaluate(evaluationContext, queryInput).map { transform(evaluationContext, it) }
     }
+
+    protected open fun transform(evaluationContext: QueryEvaluationContext, input: IStepOutput<In>): IStepOutput<Out> {
+        return transform(evaluationContext, input.value).asStepOutput(this)
+    }
+
+    @Deprecated("use the function that preserves the IStepOutput")
     abstract fun transform(evaluationContext: QueryEvaluationContext, input: In): Out
```

Comment:
Is there a need for this deprecation cycle? If that method is something that's used only internally, it could probably be dropped directly?

Reply:
I removed the annotation. Since most steps produce a new value and don't just forward the input, using that overload is still valid and reduces duplication.